### PR TITLE
feat(TFD-1713): Add the case of array of primitive data types

### PR DESCRIPTION
**Pull request by zafkirTalend**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

We generate schema of array of hierarchical data, not primitive data types.

**What is the new behavior?**

Add the case of array of primitive data types.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/149/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/149/)

File: daikon/src/main/java/org/talend/daikon/avro/inferrer/JsonSchemaInferrer.java [hunk 1]
```diff
@@ -139,9 +136,10 @@ public Schema inferSchema(String json) {
                     final ArrayNode arrayNode = (ArrayNode) nextNode;
                     Iterator<JsonNode> nodeIterator = arrayNode.elements();
                     if (nodeIterator.hasNext()) {
-                        field = new Schema.Field(mapEntry.getKey(), Schema.createArray(
-                                Schema.createRecord(getSubRecordRandomName(), null, null, false, getFields(nodeIterator.next()))),
-                                null, null, Schema.Field.Order.ASCENDING);
+                        field = new Schema.Field(mapEntry.getKey(), Schema.createArray(getAvroSchema(nodeIterator.next())), null,
+                                null, Schema.Field.Order.ASCENDING);
+                    } else {
+                        // TODO: if array field is empty, example: b:[]
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/avro/inferrer/JsonSchemaInferrer.java]: why don't you do it ?

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/165/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/165/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/168/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/168/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/169/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/169/)

**Pull request merged**